core/call_every_handler.cpp

```diff
@@ -81,6 +82,13 @@ void CallEveryHandler::run_once()
                 _entries_mutex.lock();
             }
         }
+
+        // We leave the loop.
+        // FIXME: there should be a nicer way to do this.
```

Comment:
(Just trying, maybe it's not better)

```
size_t i = 0;
while (i < _entries.size() && !_iterator_invalidated) {
    ...
    i++;
}
```

... instead of the `for` loop of line 68.

Reply:
The problem with your suggestion is that the `i++` still happens without check because during the loop while the callback is called, the remove can happen.